system_metrics_collector/test/system_metrics_collector/test_linux_memory_measurement.cpp

```diff
@@ -305,7 +305,7 @@ TEST_F(LinuxMemoryMeasurementTestFixture, TestPublishMessage)
   // check expected received message
   const auto received_message = test_receive_measurements->GetLastReceivedMessage();
 
-  EXPECT_EQ(kTestNodeName, received_message.measurement_source_name);
+  EXPECT_EQ(kTestMemoryNodeName, received_message.measurement_source_name);
```

Comment:
`kTestMemoryNodeName` is a `char[]`, so this *may* compare pointers if `received_message` is a `char *`. I suggest doing something like:
```c++
EXPECT_EQ(std::string(kTestMetricName), received_message);
```
This is working right now because `received_message` is being resolved as an `std::string` and `std::string` supports comparing against `const char *`.